app.py

```diff
@@ -37,13 +36,112 @@ def buckets() -> str:
+                            "date_modified": item["LastModified"],
+                        }
+                    )
+
+    if search_param:
+        contents = list(filter(lambda x: search_param in x['name'], contents))
+    contents = sorted(contents, key=lambda x: x["type"], reverse=True)
+    return contents
+
+
+@app.route("/search//buckets/<bucket_name>", defaults={"path": ""})
+@app.route("/search/buckets/<bucket_name>/<path:path>")
+def search_bucket(bucket_name: str, path: str) -> str:
+    s3_client = boto3.client("s3", **AWS_KWARGS)
+    responses = []
+    try:
+        marker = ''
+        while True:
+            response = s3_client.list_objects_v2(Bucket=bucket_name,
+                                                 Prefix=path,
+                                                 ContinuationToken=marker)
+            responses.append(response)
+            if not response['IsTruncated']:
+                break
+            marker = response['NextContinuationToken']
+        marker = ''
+        while True:
+            response = s3_client.list_objects_v2(Bucket=bucket_name,
+                                                 Prefix=path,
+                                                 ContinuationToken=marker,
+                                                 Delimiter="/")
+            responses.append(response)
+            if not response['IsTruncated']:
+                break
+            marker = response['NextContinuationToken']
+    except botocore.exceptions.ClientError as e:
+        match e.response["Error"]["Code"]:
+            case "AccessDenied":
+                return render_template(
+                    "error.html",
+                    error="You do not have permission to access this bucket.",
+                )
+            case "NoSuchBucket":
+                return render_template("error.html", error="The specified bucket does not exist.")
+            case _:
+                return render_template("error.html", error=f"An unknown error occurred: {e}")
+    except Exception as e:  # noqa: BLE001
+        return render_template("error.html", error=f"An unknown error occurred: {e}")
+
+    search_param = request.args['search'] if 'search' in request.args else ''
+    contents = parse_responses(responses, s3_client, bucket_name, search_param)
+    return render_template(
+        "bucket_contents.html",
+        contents=contents,
+        bucket_name=bucket_name,
+        path=path,
+        search_param=search_param,
+    )
+
+
 @app.route("/buckets/<bucket_name>", defaults={"path": ""})
 @app.route("/buckets/<bucket_name>/<path:path>")
 def view_bucket(bucket_name: str, path: str) -> str:
     s3_client = boto3.client("s3", **AWS_KWARGS)
-
+    responses = []
     try:
-        response = s3_client.list_objects_v2(Bucket=bucket_name, Prefix=path, Delimiter="/")
+        marker = ''
+        while True:
+            response = s3_client.list_objects_v2(Bucket=bucket_name,
```

Comment:
Hi, thanks for the PR! I tried to run your version on my machine but whenever I open any bucket I get "An error occurred
An unknown error occurred: An error occurred (InvalidArgument) when calling the ListObjectsV2 operation: The continuation token provided is incorrect".